include/nanoflann.hpp

```diff
@@ -855,35 +862,34 @@ class KDTreeBaseClass {
    * @param left index of the first vector
    * @param right index of the last vector
    */
-  NodePtr divideTree(Derived &obj, const IndexType left, const IndexType right,
+  NodePtr divideTree(Derived &obj, const uint64_t left, const uint64_t right,
                      BoundingBox &bbox) {
     NodePtr node = obj.pool.template allocate<Node>(); // allocate memory
 
     /* If too few exemplars remain, then make this a leaf node. */
-    if ((right - left) <= static_cast<IndexType>(obj.m_leaf_max_size)) {
+    if ((right - left) <= static_cast<uint64_t>(obj.m_leaf_max_size)) {
       node->child1 = node->child2 = NULL; /* Mark as leaf node. */
       node->node_type.lr.left = left;
       node->node_type.lr.right = right;
 
       // compute bounding-box of leaf points
       for (int i = 0; i < (DIM > 0 ? DIM : obj.dim); ++i) {
-        bbox[i].low = dataset_get(obj, obj.vind[left], i);
-        bbox[i].high = dataset_get(obj, obj.vind[left], i);
+        bbox[i].low = dataset_get(obj, obj.vAcc[left], i);
+        bbox[i].high = dataset_get(obj, obj.vAcc[left], i);
       }
-      for (IndexType k = left + 1; k < right; ++k) {
+      for (uint64_t k = left + 1; k < right; ++k) {
         for (int i = 0; i < (DIM > 0 ? DIM : obj.dim); ++i) {
-          if (bbox[i].low > dataset_get(obj, obj.vind[k], i))
-            bbox[i].low = dataset_get(obj, obj.vind[k], i);
-          if (bbox[i].high < dataset_get(obj, obj.vind[k], i))
-            bbox[i].high = dataset_get(obj, obj.vind[k], i);
+          if (bbox[i].low > dataset_get(obj, obj.vAcc[k], i))
+            bbox[i].low = dataset_get(obj, obj.vAcc[k], i);
+          if (bbox[i].high < dataset_get(obj, obj.vAcc[k], i))
+            bbox[i].high = dataset_get(obj, obj.vAcc[k], i);
         }
       }
     } else {
-      IndexType idx;
+      uint64_t idx;
```

Comment:
Same comment here, and basically, everywhere where uint64_t shows up... it lights a red light for me seen a hard-wired type instead of a typename parameter, a "using A=B;", etc. ;-)